# Botany 2021 Poster Methods
__Title__: Assembling the _Tragopogon pratensis_ genome!  
__Authors__: Kasey Khanh Pham, Xiaoxian Liu, Lucas Boatwright, Doug Soltis, Pam Soltis, Brad Barbazuk  
__Contact__: Kasey Khanh Pham at kasey.pham@ufl.edu  
__Time__: 5:00p EST, July 19th 2021  
__Conference ID__: P1GT003

## Raw Reads
DNA was extracted using a modified CTAB protocol from leaf tissue fo a single individual of _Tragopogon pratensis_ grown in the greenhouse at University of Florida, Gainesville, FL. 

Sequencing was performed on the BGI sequencing platform, generating in total ~744 million 100bp paired-end reads.

## Read Processing
### Quality Filtering and Check
BGI adaptors and poor quality reads were trimmed and filtered using [`Trimmomatic v.0.36`](http://www.usadellab.org/cms/?page=trimmomatic) using the following parameters:
```
ILLUMINACLIP:BGIadapter.fa:2:35:4:12:true
LEADING:3 TRAILING:3 SLIDINGWINDOW:5:15 MINLEN:50
```

Quality of raw reads and trimmed reads was assessed using [`FastQC`](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) and [`MultiQC`](https://multiqc.info/).

Reads from organellar genomes were identified and removed by mapping to the following reference sequences:
* _Lactuca sativa_ chloroplast genome (NCBI: [AP007232.1](https://www.ncbi.nlm.nih.gov/nuccore/AP007232.1))
* _Helianthus annuus_ chloroplast genome (NCBI: [NC_007977.1](https://www.ncbi.nlm.nih.gov/nuccore/NC_007977.1))
* _Helianthus annuus_ mitochondrial genome (NCBI: [NC_023337.1](https://www.ncbi.nlm.nih.gov/nucleotide/NC_023337.1))

[`GSNAP`](https://github.com/juliangehring/GMAP-GSNAP) was used to map filtered reads to the reference sequences using default parameters. A custom [`python 2`](https://www.python.org/) script was used to extract reads which mapped to the reference along >50% of their length with an identity score of >85%. The `filterbyname.sh` script from [`BBMap`](https://jgi.doe.gov/data-and-tools/bbtools/) was used to remove identified organellar reads.

### Assembly Preparation
Read insert size was estimated using [`SAMTools`](http://www.htslib.org/) and [`QualiMap`](http://qualimap.conesalab.org/) using default parameters for BAM files.

Genome size was estimated from k-mer frequency using [`Jellyfish`](https://github.com/gmarcais/Jellyfish/) following the procedure described by [T. Nishiyama](https://koke.asrc.kanazawa-u.ac.jp/HOWTO/kmer-genomesize.html). Several k-mer sizes were tried, with a k-mer size of 53 yielding the closest esimate (2.13 Gbp) to those estimated from flow cytometry and for the _T. dubius_ genome assembly.

[`kmergenie`](http://kmergenie.bx.psu.edu/) was used to select an optimal k-mer size for de Bruijn graph construction during assembly using the `haploid` flag, as the source plant was highly inbred and therefore homozygous at most loci. It yielded an optimal k-mer size of 85.

## Assembly
### _De-novo_ contig assembly
[`SOAPdenovo`](https://github.com/aquaskyline/SOAPdenovo2/) was used to assemble contigs from filtered reads. Default parameter values were used except for the following:
```
-K 85           # kmer size of 85
-N 2000000000   # estimated genome size of 2Gbp
-R              # resolve repeats
-F              # fill scaffolding gaps with Ns
```

Bacterial contigs were filtered out of assembly by mapping to the [RefSeq Nonredundant Prokaryotic Proteins Database](https://www.ncbi.nlm.nih.gov/refseq/about/nonredundantproteins/) using [`BLAST+`](https://blast.ncbi.nlm.nih.gov/Blast.cgi) using the following parameters:
```
-evalue 1e-5
-perc_identity .9
```

A custom `python` script was used to remove contigs which matched the bacterial database.

### Reference-based scaffolding
Contigs from _de novo_ assembly were then scaffolded against several different reference genomes:
* The _Tragopogon dubius_ draft genome assembly
* The _Lactuca sativa_ draft genome assembly ([Official Site](http://lgr.genomecenter.ucdavis.edu/))
* The _Tragopogon dubius_ draft assembly itself scaffolded against the _Lactuca sativa_ draft assembly

The scaffolding program [`RAGOO`](https://github.com/malonge/RaGOO) was used with default parameters. Unmapped contigs were separated into separate sequences using a custom `python` script and all scaffolds were filtered to a minimum length of 200bp using the `reformat.sh` script in `BBMap`.

Quality of the assembly was estimated in [`BUSCO`](https://busco.ezlab.org/) using the following parameters:
```
lineage_dataset = embryophyta_odb10
mode = genome
evalue = 1e-3
```

### Structural gene annotation
[Daren Card's gene annotation pipeline](https://gist.github.com/darencard/bb1001ac1532dd4225b030cf0cd61ce2) was followed for structural gene annotation.

Repeats in the assembly were identified _de novo_ using `RepeatModeler` in [`RepeatMasker`](http://www.repeatmasker.org). _De novo_ repeats and those identified using the Asteraceae repeat database were masked in the assembly using `RepeatMasker`.

For gene annotation, contigs less than 1000bp long were excluded using `reformat.sh` from `BBMap` due to computational limitations. The remaining contigs were first annotated in [`MAKER`](http://www.yandell-lab.org/software/maker.html) using the following resources to train the program:
* _Tragopon pratensis_ leaf and floral transcriptomes as ESTs
* _Trapogon dubius_ leaf and floral transcriptomes as alternate ESTs
* _Arabidopsis thaliana_, _Glycine max_, _Gossypium raimondii_, _Helianthus annuus_, _Lactuca sativa_, and _Solanum lycopersicum_ proteomes as protein evidence.

Gene models from `MAKER` were then provided as training data to [`SNAP`](https://github.com/KorfLab/SNAP) and [`AUGUSTUS`](http://bioinf.uni-greifswald.de/augustus/). The latter used the following parameters:
```
lineage_dataset = embryophyta_odb10
mode = genome
evalue = 1e-3
augustus_species = tomato
```
The trained models from `SNAP` and `AUGUSTUS` were used as evidence for a second round of `MAKER`. The `gff3_merge` and `fasta_merge` scripts in `MAKER` were used to extract sequences from gene model predictions made.

### Functional gene annotation
Gene functions were inferred in terms of [gene ontology](http://geneontology.org/) using [`GOMap`](https://dill-picl.org/projects/gomap/) and then tested for enrichment of GO Terms using `topGO` in [`BioConductor`](https://bioconductor.org/), using both classic count with Fisher's exact test and parent-child with Fisher's exact test. Multiple comparison testing was not performed because all GO term tests were non-significant.
