# Botany 2021 Poster Methods
__Title__: Assembling the _Tragopogon pratensis_ genome!
__Authors__: Kasey Khanh Pham, Xiaoxian Liu, Lucas Boatwright, Doug Soltis, Pam Soltis, Brad Barbazuk
__Contact__: Kasey Khanh Pham at kasey.pham@ufl.edu
__Time__: 5:00p EST, July 19th 2021
__Conference ID__: P1GT003

## Raw Reads
DNA was extracted using a modified CTAB protocol from leaf tissue fo a single individual of _Tragopogon pratensis_ grown in the greenhouse at University of Florida, Gainesville, FL. 

Sequencing was performed on the BGI sequencing platform, generating in total ~744 million 100bp paired-end reads.

## Read Processing
### Quality Filtering and Check
BGI adaptors and poor quality reads were trimmed and filtered using [`Trimmomatic v.0.36`](http://www.usadellab.org/cms/?page=trimmomatic) using the following parameters:
```
ILLUMINACLIP:BGIadapter.fa:2:35:4:12:true
LEADING:3 TRAILING:3 SLIDINGWINDOW:5:15 MINLEN:50
```

Quality of raw reads and trimmed reads was assessed using [`FastQC`](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) and [`MultiQC`](https://multiqc.info/).

Reads from organellar genomes were identified and removed by mapping to the following reference sequences:
* _Lactuca sativa_ chloroplast genome (NCBI: [AP007232.1](https://www.ncbi.nlm.nih.gov/nuccore/AP007232.1))
* _Helianthus annuus_ chloroplast genome (NCBI: [NC_007977.1](https://www.ncbi.nlm.nih.gov/nuccore/NC_007977.1))
* _Helianthus annuus_ mitochondrial genome (NCBI: [NC_023337.1](https://www.ncbi.nlm.nih.gov/nucleotide/NC_023337.1))

[`GSNAP`](https://github.com/juliangehring/GMAP-GSNAP) was used to map filtered reads to the reference sequences using default parameters. A custom [`python 2`](https://www.python.org/) script was used to extract reads which mapped to the reference along >50% of their length with an identity score of >85%. The `filterbyname.sh` script from [`BBMap`](https://jgi.doe.gov/data-and-tools/bbtools/) was used to remove identified organellar reads.

### Assembly Preparation
Read insert size was estimated using [`SAMTools`](http://www.htslib.org/) and [`QualiMap`](http://qualimap.conesalab.org/) using default parameters for BAM files.

Genome size was estimated from k-mer frequency using [`Jellyfish`](https://github.com/gmarcais/Jellyfish/) following the procedure described by [T. Nishiyama](https://koke.asrc.kanazawa-u.ac.jp/HOWTO/kmer-genomesize.html). Several k-mer sizes were tried, with a k-mer size of 53 yielding the closest esimate (2.13 Gbp) to those estimated from flow cytometry and for the _T. dubius_ genome assembly.

[`kmergenie`](http://kmergenie.bx.psu.edu/) was used to select an optimal k-mer size for de Bruijn graph construction during assembly using the `haploid` flag, as the source plant was highly inbred and therefore homozygous at most loci. It yielded an optimal k-mer size of 85.

## Assembly
### De-novo contig assembly

